### PR TITLE
MLX provider: Nameserver support (stage 1/3: MPI-CH3 is covered)

### PR DESCRIPTION
**Pull request by alexander-sannikov**
Implemented built-in NS support in MLX provider.
MPICH-CH3 usage scenario is covered.
 
Signed-off-by: Sannikov, Alexander <alexander.sannikov@intel.com>

### CODE REVIEW

File: prov/mlx/src/mlx_av.c [hunk 1]
```diff
@@ -116,9 +162,14 @@ static int mlx_av_insert(
 		} else {
 			if (av->eq) {
 				mlx_av_write_event( av, i,
-						MLX_TRANSLATE_ERRCODE(status),
-						context);
+					MLX_TRANSLATE_ERRCODE(status),
+					context);
 			}
+			break;
```
Comment (shefty) [inline prov/mlx/src/mlx_av.c]: If we take this path, we leak ep_params.address

File: prov/mlx/src/mlx_callbacks.c [hunk 2]
```diff
@@ -1,3 +1,6 @@
+/* -*- mode: c; c-basic-offset: 8; -*-
+ * vim: noexpandtab sw=8 ts=8 sts=0:
+ */
```
Comment (shefty) [inline prov/mlx/src/mlx_callbacks.c]: Please drop these changes from the files.

File: prov/mlx/src/mlx_fabric.c [hunk 3]
```diff
@@ -88,5 +171,37 @@ int mlx_fabric_open(
...
+
+		status = ofi_ns_init(&ns_attr,
+				  &mlx_descriptor.name_serv);
+		if (status) {
+			FI_INFO(&mlx_prov, FI_LOG_CORE,
+			        "ofi_ns_init returns %d\n", status);
+			free(fabric_priv);
+			return status;
+		}
+		ofi_ns_start_server(&mlx_descriptor.name_serv);
```
Comment (shefty) [inline prov/mlx/src/mlx_fabric.c]: This is a large segment of code that would be better isolated into a new function.

**Comment by alexander-sannikov** [comment]
@shefty I've fix all your remarks.

File: prov/mlx/src/mlx_av.c (hunk 1 above)
Comment (alexander-sannikov) [reply prov/mlx/src/mlx_av.c]: Now free calls before break. Problem should go away.

**Comment by lanl-ompi** [comment]
Can one of the admins verify this patch?

**Comment by hppritcha** [comment]
add to whitelist

**Pull request merged**